File: backend/src/main/java/com/pk/cruncher/entity/RecipeIngredient.java
package com.pk.cruncher.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@Entity
public class RecipeIngredient extends BaseEntity {

    @ManyToOne
    @JoinColumn(name = "structure_id")
    private Item structureItem;

    @ManyToOne
    @JoinColumn(name = "ingredient_id")
    private Item ingredientItem;

    private Double grossQuantity;

    private Double netQuantity;

    private Boolean lock;

    private Boolean cleared;

    private Boolean boiled;

    private Boolean fried;

    private Boolean stewed;

    private Boolean baked;
}

File: src/app/model/recipe-ingredient.ts
import {Ingredient} from './ingredient';

export interface RecipeIngredient {
  id: string | null;
  ingredient: Ingredient | null;
  grossQuantity: number;
  netQuantity: number;
  lock: boolean;
  cleared: boolean;
  boiled: boolean;
  fried: boolean;
  stewed: boolean;
  baked: boolean;
}
